src/pgduckdb_ddl.cpp

```diff
@@ -187,18 +190,19 @@ duckdb_create_table_trigger(PG_FUNCTION_ARGS) {
 	}
 
 	if (SPI_processed != 1) {
-		elog(ERROR, "Expected single table to be created, but found %" PRIu64, SPI_processed);
+		throw std::runtime_error("Expected single table to be created, but found " + std::to_string(SPI_processed));
 	}
 
 	HeapTuple tuple = SPI_tuptable->vals[0];
 	bool isnull;
 	Datum relid_datum = SPI_getbinval(tuple, SPI_tuptable->tupdesc, 1, &isnull);
 	if (isnull) {
-		elog(ERROR, "Expected relid to be returned, but found NULL");
+		throw std::runtime_error("Expected relid to be returned, but found NULL");
 	}
+
 	Datum is_temporary_datum = SPI_getbinval(tuple, SPI_tuptable->tupdesc, 2, &isnull);
 	if (isnull) {
-		elog(ERROR, "Expected temporary boolean to be returned, but found NULL");
+		throw std::runtime_error("Expected temporary boolean to be returned, but found NULL");
```

Comment:
I think we discussed this in person, but I'll write it down here as well:

I'm not entirely convinced yet that changing all the elogs to exceptions in this file and in `pgduckdb_ruleutils.cpp` is all that helpful. It doesn't hurt either functionally, because you've wrapped this fuction in an `InvokeCPPFunc` call, and there's no `PG_TRY` around them. So there is no problem with throwing exceptions now.

But I think these specific functions probably won't benefit from being rewritten to C++ completely, and would be easier to maintain in Postgres C style. Mainly because they call a lot of Postgres functions and very few DuckDB functions. Wrapping all of those Postgres calls in `PostgresFunctionGuard` calls doesn't necessarily sound like it makes the code easier to read and maintain. I'm happily proven wrong though. 

If you're fairly sure this is the right direction for these functions I'm fine with merging this now. But I think it should be a separate commit from most of the other changes (specifically the new and improved macros). To make reverting it easier if it turns out we want to make these functions fully Postgres C instead of fully C++.